schemas/fig.d.ts

```diff
@@ -105,6 +105,30 @@ declare namespace Fig {
      * The "-" suggestion is hidden in the `cd` spec. You will only see it if you type cd -
      */
     hidden?: boolean;
+
+    /**
+     *
+     * Specifies whether a suggestion is deprecated.
+     * It is possible to optionally specify a deprecation message and an alternative command to run.
+     *
+     * `deprecated: true` and `deprecated: { }` will just display the option as deprecated.
+     *
+     */
+    deprecated?:
+      | boolean
+      | {
+          /**
+           * Provide an alternative to the deprecated command.
+           *
+           * @example
+           * alternative: 'npm install'
+           */
+          alternative?: string;
```

Comment:
I like this idea a lot :) One thing that might help Fig be even more useful here in the future is if we type this as
```suggestion
          alternative?: Omit<BaseSuggestion, "deprecated">;
```
Which adds a little more structure/modularity to the alternative so we could (in the future) add an option in settings to replace deprecated commands and flags with their alternative if they exist, e.g. in the `docker-compose` change you made we could just insert `--ansi never` when the user selects the `--no-ansi` suggestion. This probably wouldn't be great default behavior in case you _really_ want to insert the original suggestion, but could be a nice opt in feature, and makes the spec standard a little more information rich.

Reply:
A possible edge case where that structure is not applicable is for example if an option has moved to a subcommand:
e.g.
`foo bar --remove` -> `foo removebar`
I recognize it is a strange one, but
`alternative?: Omit<BaseSuggestion, "deprecated"> | string` seems probably better. If alt is a string just a warning message is displayed.

Reply:
If the intent of the `alternative` string is to offer replacement text, I think anything we could accomplish with a string could be accomplished with the `insertValue` in `BaseSuggestion`, as this is already a string, right? If it is purely informational, then we could use the `description` field of the `BaseSuggestion` but ultimately this decision could be left up to whoever is implementing the spec for that particular deprecated object.

Reply:
It makes more sense now!